backend/models/loginRecord.js

```diff
@@ -0,0 +1,127 @@
+/**
+ *	Copyright (C) 2014 3D Repo Ltd
+ *
+ *	This program is free software: you can redistribute it and/or modify
+ *	it under the terms of the GNU Affero General Public License as
+ *	published by the Free Software Foundation, either version 3 of the
+ *	License, or (at your option) any later version.
+ *
+ *	This program is distributed in the hope that it will be useful,
+ *	but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *	GNU Affero General Public License for more details.
+ *
+ *	You should have received a copy of the GNU Affero General Public License
+ *	along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+ "use strict";
+ const db = require("../handler/db");
+ 
+ //detects edge as browser but not device
+ var uaParserJs = require('ua-parser-js');
+ var device = require('device');
+ var ip2location = require('ip-to-location');
+
+ const LoginRecord = {};
+ 
+ LoginRecord.saveLoginRecord = async (req) => {
+        var loginRecord = {
+                _id : req.sessionID,
+                loginTime : new Date(),
+                ipAddr: req.ips[0] || req.ip                    
+        }
+
+        const referrer = req.header('Referer');
+        if(referrer != null){
+                loginRecord.referrer = referrer;
+        }
+
+        const location = await getLocationFromIPAddress(loginRecord.ipAddr);   
```

Comment:
It's a bit here or there, but you could use destructuring assignments here (and people seem to prefer this in modern javascript programming)
```suggestion
        const {country_name, city} = await getLocationFromIPAddress(loginRecord.ipAddr);   
        loginRecord.location = {
                country: country_name,
                city
        };
        
```